test/functional/fixtures/timeout-worker.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var ponos = require('../../../');
+var Promise = require('bluebird');
+
+var TaskFatalError = ponos.TaskFatalError;
+
+/**
+ * A worker that will publish a message to an event emitter. This worker also
+ * will take a long time to do so (causing timeouts).
+ * @param {object} job Object describing the job.
+ * @param {string} job.queue Queue on which the message will be published.
+ * @returns {promise} Resolved when the message is put on the queue.
+ */
+module.exports = function (job) {
+  return Promise.resolve()
+    .then(function () {
+      if (!job.eventName) { throw new TaskFatalError('eventName is required'); }
+      if (!job.message) { throw new TaskFatalError('message is required'); }
+    })
+    .then(function () {
+      var timeout = module.exports.globalTimeout;
+      module.exports.globalTimeout = module.exports.globalTimeout / 2;
```

Comment:
yes, for this particular example worker. I'm thinking now that maybe I should supply a list of timeouts, and once there are no more timeouts, it throws a `TaskFatalError`. That could be useful...